packages/core/src/components/hotkeys/hotkeysDialog.tsx

```diff
@@ -59,8 +66,18 @@ class HotkeysDialog {
         this.hotkeysQueue.push(hotkeys);
 
         // reset timeout for debounce
-        clearTimeout(this.timeoutToken);
-        this.timeoutToken = setTimeout(this.show, 10);
+        clearTimeout(this.showTimeoutToken);
+        this.showTimeoutToken = setTimeout(this.show, DELAY_IN_MS);
+    }
+
+    /**
+     * Since one hotkey toggles the dialog open and closed, we need to
+     * introduce a delay to ensure that all hotkey listeners fire with the
+     * dialog in a consistent state.
```

Comment:
could also just edit this text so it's useful to a consumer, to explain why you'd want to call this function instead of regular `hide()`.